### PR TITLE
[Compute] sig image-definition create: add --os-state to allow specifying whether the virtual machines created under this image are 'Generalized' or 'Specialized'; sig image-definition create: add --hyper-v-generation to allow specifying the hypervisor generation

### PR DESCRIPTION
**Pull request by qwordy**
Issue #11117

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@bim-msft  please help review.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -756,7 +756,7 @@ def load_arguments(self, _):
         c.argument('sku', options_list=['--sku', '-s'], help='image sku')
         c.argument('publisher', options_list=['--publisher', '-p'], help='image publisher')
         c.argument('os_type', arg_type=get_enum_type(['Windows', 'Linux']), help='the type of the OS that is included in the disk if creating a VM from user-image or a specialized VHD')
-        c.ignore('os_state')  # service is not ready
+        c.argument('os_state', arg_type=get_enum_type(self.get_models('OperatingSystemStateTypes')), help="This property allows the user to specify whether the virtual machines created under this image are 'Generalized' or 'Specialized'.")
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py]: Is there any link about this argument? What's the meaning of "Generalized" and "Specialized"?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=%2Fsrc%2FCAPS%2FDoc%2FPrivateGalleryDesignAndApiDoc.md&version=GBmaster

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: I copy the help text from swagger.

**Pull request merged**